lib/rspec/matchers/composable.rb

```diff
@@ -106,7 +106,9 @@ def surface_descriptions_in(item)
         elsif enumerable?(item)
           begin
             item.map { |subitem| surface_descriptions_in(subitem) }
-          rescue IOError # STDOUT is enumerable but `map` raises an error
+          # STDOUT is enumerable but `map` raises an error
+          # Range is enumerable except when infinite
+          rescue IOError, TypeError
```

Comment:
Instead of rescueing `TypeError`, should we consider changing the definition of `enumerable?` so that ranges aren't enumerated?  After all, even for an enumerable range (e.g. all ints: `1..10`), I don't think we want it enumerated in the description.  We'd rather just have it inspected and displayed as `1..10`, right?

If we go that route, we should perhaps rename `enumerable?` to `should_enumerate?` or something.